build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```diff
@@ -318,3 +319,11 @@ fun TaskContainer.registerCITestDistributionLifecycleTasks() {
         group = ciGroup
     }
 }
+
+// https://github.com/gradle/gradle-private/issues/3380
+fun Test.configureAndroidUserHome() {
+    doFirst {
```

Comment:
Let's not do this as a doFirst. Can't we have some location for each test task which is not the `temporaryDir`? We already have locations for the whole test files, can't we add it to the side there?